docs/spec/glossary.rst

```diff
@@ -12,18 +12,101 @@ This section defines a few terms that may be used elsewhere in the specification
       :term:`type expression`, sometimes with additional :term:`type qualifiers <type qualifier>`.
       See :ref:`"Type and annotation expression" <annotation-expression>` for details.
 
+   assignable
+      If a type ``B`` is "assignable to" a type ``A``, a  type checker should
+      not error on the assignment ``x: A = b``, where ``b`` is some expression
+      whose type is ``B``. Similarly for function calls and returns: ``f(b)``
+      where ``def f(x: A): ...`` and ``return b`` inside ``def f(...) -> A:``
+      are both valid (not a type error) if and only if ``B`` is assignable to
+      ``A``. In this case ``A`` is "assignable from" ``B``. For :term:`fully
+      static types <fully static type>`, "assignable to" is equivalent to
+      ":term:`subtype` of" and "assignable from" is equivalent to
+      ":term:`supertype` of". For :term:`gradual types <gradual type>`, a type
+      ``B`` is assignable to a type ``A`` if there exist fully static
+      :term:`materializations <materialize>` ``A'`` and ``B'`` of ``A`` and
+      ``B``, respectively, such that ``B'`` is a subtype of ``A'``. See
+      :ref:`type-system-concepts`.
+
+   consistent
+      Two :term:`fully static types <fully static type>` are "consistent with"
+      each other if they are the same type. Two gradual types are "consistent
+      with" each other if they could :term:`materialize` to the same type. See
+      :ref:`type-system-concepts`. If two types are consistent, they are both
+      :term:`assignable` to and from each other.
+
+   consistent subtype
+      "Consistent subtype" is synonymous with ":term:`assignable` to" (and
+      "consistent supertype" is synonymous with "assignable from"). See
+      :ref:`type-system-concepts`.
+
    distribution
       The packaged file which is used to publish and distribute
       a release. (:pep:`426`)
 
+   equivalent
+      Two :term:`fully static types <fully static type>` ``A`` and ``B`` are
+      equivalent if ``A`` is a :term:`subtype` of ``B`` and ``B`` is a
+      :term:`subtype` of ``A``. This implies that ``A`` and ``B`` represent the
+      same set of possible runtime objects.
```

Comment:
In concepts.rst you also define the equivalence relation for gradual types; that should be added here. It is important if `assert_type()` is defined in terms of equivalence.